src/sql-parser/src/parser.rs

```diff
@@ -1981,8 +1986,13 @@ impl Parser {
         }
     }
 
-    /// Parse a SQL datatype (in the context of a CREATE TABLE statement for example)
-    fn parse_data_type(&mut self) -> Result<DataType, ParserError> {
+    /// Parse a SQL datatype (in the context of a CREATE TABLE statement for
+    /// example)
+    ///
+    /// # Arguments
+    /// - `allow_vector` specifies whether the caller accepts vector data types,
+    ///   e.g. `INT LIST`, and in the future `INT ARRAY`.
+    fn parse_data_type(&mut self, allow_vector: bool) -> Result<DataType, ParserError> {
```

Comment:
I'm not sure what problem `allow_vector` is solving now? The test suite passes with flying colors without it, and I don't see any reason why we shouldn't allow syntax like `SELECT INT LIST '{1, 2}'` since it seems we can parse it ok. (Unlike with `int[]` where we explicitly want to be PostgreSQL compatible, we've already decided to jettison compatibility with the list type, so I think we're free to do the most useful thing, rather than the most PostgreSQL thing.)

Reply:
Sorry, Jamii's comment that I'm working from...

> The api in future PRs will mimic the sensible parts of the postgres ARRAY api.

...is ambiguous. I figured we wanted to make `LIST` as `ARRAY`-like as possible, but it sounds like this is not the case.

Reply:
Ah yeah, that's not very well spec'd out :D.